example-app/App.tsx

```diff
@@ -65,11 +96,28 @@ const App = () => {
         <View style={backgroundStyle}>
           <Button title="Perform action" disabled={true} />
         </View>
+        {result && (
+          <View>
+            <Text>
+              {result.success ? 'SUCCESS' : `FAILURE: ${result.errorReason}`}
+            </Text>
+          </View>
+        )}
       </ScrollView>
     </SafeAreaView>
   );
 };
 
+type DeepLinkResult =
```

Comment:
Better to define at the top, before it is used and where you would expect to have type definitions